### PR TITLE
rust: run stack image as non root

### PR DESCRIPTION
**Pull request by skoh7645**
### Checklist:

- [x] Read the [Code of Conduct](https://github.com/appsody/website/blob/master/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/appsody/website/blob/master/CONTRIBUTING.md).

- [x] Followed the [commit message guidelines](https://github.com/appsody/website/blob/master/CONTRIBUTING.md#commit-message-guidelines).

- [x] Stack adheres to [Appsody stack structure](https://github.com/appsody/website/blob/master/content/docs/stacks/stacks-overview.md#stack-structure).

### Modifying an existing stack:

- [x] Updated the stack version in `stack.yaml`

Run `rust` stack as rust user rather than root

### Related Issues:
Fixes: #530

### CODE REVIEW

**Comment by skoh7645** [comment]
I tested this on an Ubuntu VM and it solved #530 for me. @slinkydeveloper does this fix the issue for you?

File: experimental/rust/image/Dockerfile-stack [hunk 1]
```diff
@@ -20,6 +20,9 @@ ENV APPSODY_DEBUG="cargo build && lldb-server platform --listen '*:1234' --min-g
 ENV APPSODY_DEBUG_ON_CHANGE="$APPSODY_DEBUG"
 ENV APPSODY_DEBUG_KILL=true
 
+RUN useradd rust
+USER rust
+
 COPY ./LICENSE /licenses
 COPY ./project /project
 COPY ./config /config
```
Comment (EnriqueL8) [inline experimental/rust/image/Dockerfile-stack]: Should we also add `--chown` to the copy command?

Comment (skoh7645) [reply experimental/rust/image/Dockerfile-stack]: can do

File: experimental/rust/image/Dockerfile-stack [hunk 2]
```diff
@@ -20,9 +20,12 @@ ENV APPSODY_DEBUG="cargo build && lldb-server platform --listen '*:1234' --min-g
...
 
-COPY ./LICENSE /licenses
-COPY ./project /project
-COPY ./config /config
+RUN useradd rust
+USER rust
+
+COPY --chown=rust:rust ./LICENSE /licenses
+COPY --chown=rust:rust ./project /project
+COPY --chown=rust:rust ./config /config
```
Comment (neeraj-laad) [inline experimental/rust/image/Dockerfile-stack]: copy --chown is not supported in older docker versions. We should add the requirement for that in stack.yaml.

**Comment by slinkydeveloper** [comment]
Yes it fixes the issue. Btw i've tried `appsody stack validate` works on my machine

File: experimental/rust/image/Dockerfile-stack [hunk 3]
```diff
@@ -4,6 +4,17 @@ RUN apt-get update && apt-get install -y lldb
 
 ENV CARGO_HOME=/usr/local/cargo/deps
 
+RUN  groupadd --gid 1000 rust_group \
+  && useradd --uid 1000 --gid rust_group --shell /bin/bash --create-home rust_user \
```
Comment (Kamran64) [inline experimental/rust/image/Dockerfile-stack]: Should this be rust_user?

**Comment by skoh7645** [comment]
Closing this for now as we are prioritising the incubator stacks. Will reopen later

**Pull request closed without merging**